finality-aleph/src/validator_network/heartbeat.rs

```diff
@@ -0,0 +1,75 @@
+use codec::{Decode, Encode};
+use tokio::{
+    io::{AsyncRead, AsyncWrite},
+    time::{sleep, timeout, Duration},
+};
+
+use crate::validator_network::io::{receive_data, send_data};
+
+const HEARTBEAT_TIMEOUT: Duration = Duration::from_secs(5);
+const MAX_MISSED_HEARTBEATS: u32 = 4;
+
+/// Represents the heartbeat message. Holds a single integer,
+/// which should be interpreted as a message counter.
+#[derive(Debug, Clone, Encode, Decode)]
+struct Heartbeat(u32);
+
+/// Sends heartbeat messages at regular intervals, indefinitely.
+/// Fails if the communication channel is closed.
+pub async fn heartbeat_sender<S: AsyncWrite + Unpin + Send>(mut stream: S) {
+    let mut counter: u32 = 0;
+    loop {
+        stream = match send_data(stream, Heartbeat(counter)).await {
```

Comment:
What is this counter for? It is ignored on the other side of the stream. Maybe we just could make it something constant without incrementing?